### PR TITLE
Task/geoprocessing 24 remove python2 things

### LINKED ISSUE #24: Remove python2 compatibility things from pygeoprocessing
Pygeoprocessing still has a few things left over from when we were trying to have the package be cross-compatible with `python2` and `python3`, things like the `future` dependency and a few conditional imports (like `Queue` being in different places).  Now that we're python3-only, we should clean these things up.

Known parts to this:
- [x] `mock` --> `unittest.mock`
- [x] Remove `future` from `requirements.txt`
- [x] Remove `future` and `future`-defined imports (`builtins`, others?) throughout the package

### PR DESCRIPTION
**Pull request by richpsharp**
Removed all Python 2 patch things, updated docstrings as needed, cleaned up imports, and fixed any open pep8lint issues. No need to update HISTORY since none of these changes are user facing.

closes #24

### CODE REVIEW

**Comment by richpsharp** [comment]
@phargogh, I put you as a reviewer on this since you're the one that made all the Python 2/3 compatible things back in the day and would have a better eye for it. Also I'm merging this into the 2.0 release branch because of all the docstring changes. I *thought* we already changed all the "Parameters" to "Args" but that's not showing up and I'm suspicious so I wanted to make sure it was here. There's no other "non-backwards-compatible" issues though.

File: tests/test_geoprocessing.py [hunk 1]
```diff
@@ -110,30 +111,13 @@ def test_star_import(self):
...
-        import pygeoprocessing
-
-        with unittest.mock.patch(
-                        'pygeoprocessing.pkg_resources.get_distribution',
-                        side_effect=DistributionNotFound('pygeoprocessing')):
-            with self.assertRaises(RuntimeError):
-                # RuntimeError is a side effect of `import pygeoprocessing`,
-                # so we reload it to retrigger the metadata load.
-                pygeoprocessing = importlib.reload(pygeoprocessing)
-
```
Comment (phargogh) [inline tests/test_geoprocessing.py]: I remember we had talked about removing this, but it still looks like that complicated versioning stuff I had added to `__init__.py` is still in there.  Should it be removed from `__init__.py`?

File: tests/test_geoprocessing.py [hunk 2]
```diff
@@ -79,7 +82,6 @@ def _read_raster_to_array(raster_path):
 
 class PyGeoprocessing10(unittest.TestCase):
     """Tests for the PyGeoprocesing 1.0 refactor."""
-
```
Comment (phargogh) [inline tests/test_geoprocessing.py]: This is trivial, but doesn't PEP8 call for a blank line here?

**Review by phargogh — CHANGES_REQUESTED** [review]
Just a couple of comments for you, but it overall looks good to me!

File: tests/test_geoprocessing.py (hunk 1 above)
Comment (richpsharp) [reply tests/test_geoprocessing.py]: Yes I forgot about those! Patched in 8c02c921416332cfd8b655d89abd3cf9632fcf75

File: tests/test_geoprocessing.py (hunk 2 above)
Comment (richpsharp) [reply tests/test_geoprocessing.py]: I looked briefly and couldn't see where PEP8 calls for that, plus my linter didn't say anything, but it didn't say anything when I added a blank line either, so patched in d8f5fcf2df1cfa19d679ed8fb21d295a56ffe234

**Comment by richpsharp** [comment]
hi @phargogh i made the changes you requested and re-requested your review

**Review by phargogh — APPROVED** [review]
Yep, looks good to me!

**Comment by phargogh** [comment]
Whoah!  OK @richpsharp @davemfish @dcdenu4 I was just able to resolve a merge conflict myself and not have write access to @richpsharp 's fork.  I'm sure you guys knew about this, but it was quite the surprise to me!

**Pull request merged**